src/main/java/com/notably/logic/suggestion/commands/OpenSuggestionCommand.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Override
+    public void execute(Model model) {
+        String responseText = "Open a note";
+        model.setResponseText(responseText);
+        System.out.println(responseText);
+
+        List<Path> possiblePaths = getPossiblePaths(path);
+        List<SuggestionItem> suggestions = getSuggestions(possiblePaths, model);
+
+        for (SuggestionItem suggestionItem : suggestions) {
+            System.out.println(suggestionItem.getDisplayText());
+        }
+
+        model.setSuggestions(suggestions);
+    }
+
+    private List<Path> getPossiblePaths(Path path) {
+        List<Path> possiblePaths = new ArrayList<>();
+        Node<String> root = null;
+
+        // TODO: traverse the data structure. For now will use dummy values.
+        if (path instanceof AbsolutePath) { // Search from root.
+            root = createTree();
+        } else {
+            // TODO
+        }
+
+        List<String> components = path.getComponents();
+        List<Node<String>> children = new ArrayList<>();
+        int index = 0;
+        Queue<Node<String>> queue = new LinkedList<>();
+        for (Node<String> rootChild : root.getChildren()) { //don't need add root to queue
+            queue.add(rootChild);
+        }
+
+        while (!queue.isEmpty()) {
+            Node<String> curr = queue.poll();
+            String dir = components.get(index);
+            if (curr.getTitle().equals(dir)) {
+                children = curr.getChildren();
+                if (index == components.size() - 1) { // just get all the children
+                    break;
+                } else {
+                    queue.clear();
+                    for (Node<String> child : children) {
+                        queue.add(child);
+                    }
+                    index++;
+                }
+            }
+        }
+
+        if (children.isEmpty()) { // if the note doesn't have any children
+            Path newPath = AbsolutePath.fromComponent(components);
+            possiblePaths.add(newPath);
+        } else {
+            for (Node<String> child : children) {
+                List<String> newComponents = new ArrayList<>();
+                newComponents.addAll(components);
+                newComponents.add(child.getTitle());
+                Path newPath = AbsolutePath.fromComponent(newComponents);
+                possiblePaths.add(newPath);
+            }
+        }
+        return possiblePaths;
+    }
+
+    private List<SuggestionItem> getSuggestions(List<Path> possiblePaths, Model model) {
+        List<SuggestionItem> suggestions = new ArrayList<>();
+        for (Path path : possiblePaths) {
+            String displayText = path.toString();
+            Runnable action = () -> {
+                model.setInput(displayText);
+            };
+            SuggestionItem suggestionItem = new SuggestionItemImpl(displayText, action);
+            suggestions.add(suggestionItem);
+        }
+
+        return suggestions;
```

Comment:
Take a look at `Stream#map` [here](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Stream.html#map(java.util.function.Function)) to make your code above more [declarative](https://en.wikipedia.org/wiki/Declarative_programming)!

For this case, the above code can be rewritten as:
```java
return possiblePaths.stream()
  .map(path -> {
    String displayText = path.toString();
    Runnable action = () -> {
      model.setInput(displayText);
    };
    return new SuggestionItemImpl(displayText, action);
  })
  .collect(Collectors.toList());
```

Reply:
Wowza sickerino! Thanks kevin! I updated the code.